### PR TITLE
[mlir][spirv] Adapt downstream `client-api` option patch after upstreaming

### PR DESCRIPTION
**Pull request by victor-eds**
d76fffc introduced a new option to `-convert-spirv-to-llvm`, upstreamed by https://github.com/llvm/llvm-project/commit/12ce9fd1248c6321b343601c1a2468ac7e00c9da. Adapt our code to avoid future merge conflicts.

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]

File: mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ using namespace mlir;
 // Constants
 //===----------------------------------------------------------------------===//
 
-static constexpr unsigned defaultAddressSpace = 0;
+constexpr unsigned defaultAddressSpace = 0;
```
Comment (etiotto) [inline mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:36]: Why removing "static" ? This global variable is used within this file only.

**Review by etiotto — APPROVED** [review]

**Review by Naghasan — APPROVED** [review]

File: mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp (hunk 1 above)
Comment (victor-eds) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:36]: `constexpr` already implies static linkage

Comment (victor-eds) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:36]: It was nitpicking, also, other `static constexpr` can be found all around the file

Comment (etiotto) [reply mlir/lib/Conversion/SPIRVToLLVM/SPIRVToLLVM.cpp:36]: ok, somehow I forgot about that.  I always prefer to use "static" in situation like this one just to make it explicit.

**Pull request merged**